File: integrations/aws-strands/error-code-corpus/typescript/sibling_same_named_locals.ts
/** Two sibling branches binding the same local name to different codes. */
import { EventType, type BaseEvent } from "@ag-ui/core";

export function emit(kind: string): BaseEvent | undefined {
  if (kind === "first") {
    const failureCode = "FIRST_SIBLING";
    return {
      type: EventType.RUN_ERROR,
      message: "The first sibling.",
      code: failureCode,
    };
  }
  if (kind === "second") {
    const failureCode = "SECOND_SIBLING";
    return {
      type: EventType.RUN_ERROR,
      message: "The second sibling.",
      code: failureCode,
    };
  }
  return undefined;
}

File: integrations/aws-strands/error-code-corpus/python/sibling_same_named_locals.py
"""Two sibling branches binding the same local name to different codes."""

from ag_ui.core import EventType, RunErrorEvent


def emit(kind: str) -> RunErrorEvent | None:
    if kind == "first":
        failure_code = "FIRST_SIBLING"
        return RunErrorEvent(
            type=EventType.RUN_ERROR,
            message="The first sibling.",
            code=failure_code,
        )
    if kind == "second":
        failure_code = "SECOND_SIBLING"
        return RunErrorEvent(
            type=EventType.RUN_ERROR,
            message="The second sibling.",
            code=failure_code,
        )
    return None
